### ISSUE TITLE
axios.getUri doesn't use baseURL

### ISSUE THREAD

**Issue by friedrith** [axios/axios#2468]
**Describe the bug**
When using https://github.com/axios/axios/pull/1712 example and adding baseURL in config, baseURL seems to not be used.

**To Reproduce**

```js
const axios = require("axios");
const fakeConfig = {
  method: "post",
  url: "/user/12345",
  baseURL: "/foo"
};
console.log(axios.getUri(fakeConfig));
```

**Expected behavior**
`axios.getUri` should return `/foo/user/12345`.

If this expected behavior seems right, I can work on a fix.

**Environment:**
 - Axios Version: 0.19.0
 - OS: Linux, Windows
 - Browser Chrome but also happen on Firefox and node

**Comment by yasuf** [comment]
Based on the definition of URI from [this article](https://dev.to/flippedcoding/what-is-the-difference-between-a-uri-and-a-url-4455) getting just `/user/12345` is a good URI, since it identifies what's being requested even though it doesn't locate the exact endpoint, if that's the case then `getUri` is working as expected. Let me know if you find a different definition!

**Comment by friedrith** [comment]
URI may be only `/user/12345` if you want to to identify the request locally to a endpoint. If you want to identifiy the request globally the endpoint URI should be including IHMO. URI specification shows examples even including the protocole https://tools.ietf.org/html/rfc3986.

**Comment by yasuf** [comment]
now that we're going to modify it, from the looks of it the URI should contain a full resource, so it might need to include the domain, etc from where the request is being made, e.g. `https://github.com/foo/user/12345`, might as well add that to this method

**Comment by friedrith** [comment]
I agree. Depending on what is provided into the config, the method should return as much information as possible including protocol, domain, port, path, anchor, query, etc.

**Comment by Alanscut** [comment]
Fixed in #2555

**Comment by sakarit** [comment]
PR #2555 was never merged so this should still be open. The solution looks valid though, so I guess a new PR could be opened. I will try to do that later.

**Comment by jteppinette** [comment]
There is a ton of documentation on the Axios website regarding `baseURL` and none of it has been working for months.

**Comment by pfdgithub** [comment]
any progress?

**Comment by jasonsaayman** [comment]
Fixed by #3737
